packages/swc-configs/tests/dev.test.ts

```diff
@@ -1,56 +1,115 @@
-import type { Config } from "@swc/core";
-import { DefaultDevConfig, defineDevConfig } from "../src/dev.ts";
+import type { Config, EsParserConfig, TsParserConfig } from "@swc/core";
+import type { SwcConfigTransformer } from "../src/applyTransformers.ts";
+import { defineDevConfig } from "../src/dev.ts";
 
-test("when no options are provided, return the default config", () => {
-    const result = defineDevConfig();
+const Browsers = ["last 2 versions"];
 
-    expect(result).toEqual(DefaultDevConfig);
+test("provided browsers are set as \"env.targets\"", () => {
+    const result = defineDevConfig({
+        browsers: Browsers
+    });
+
+    expect(result.env?.targets).toBe(Browsers);
 });
 
-test("when \"fastRefresh\" is true, the fast refresh configuration is included", () => {
+test("when fastRefresh is true, react refresh is true", () => {
     const result = defineDevConfig({
+        browsers: Browsers,
         fastRefresh: true
     });
 
-    expect(result).toMatchSnapshot();
+    expect(result.jsc?.transform?.react?.refresh).toBeTruthy();
+});
+
+test("when fastRefresh is false, react refresh is false", () => {
+    const result = defineDevConfig({
+        browsers: Browsers,
+        fastRefresh: false
+    });
+
+    expect(result.jsc?.transform?.react?.refresh).toBeFalsy();
 });
 
-test("when \"parser\" is \"ecmascript\", the configuration parser is ecmascript", () => {
+test("when parser is \"ecmascript\", the configuration parser is ecmascript", () => {
     const result = defineDevConfig({
+        browsers: Browsers,
         parser: "ecmascript"
     });
 
-    expect(result).toMatchSnapshot();
+    expect(result.jsc?.parser?.syntax).toBe("ecmascript");
 });
 
-test("when a config override function is provided, the function argument is the config with the non-config override options applied", () => {
-    const expectedArgument = defineDevConfig({
-        fastRefresh: true
+test("when parser is \"ecmascript\", jsx parsing is enabled", () => {
+    const result = defineDevConfig({
+        browsers: Browsers,
+        parser: "ecmascript"
     });
 
-    const fct = jest.fn<Config, [Config]>(() => ({
-        jsc: {
-            parser: {
-                syntax: "ecmascript"
-            }
-        }
-    }));
+    expect((result.jsc?.parser as EsParserConfig).jsx).toBeTruthy();
```

Comment:
missing a ? after parser the cast? parser could be null i think

Reply:
Will never happen for this test.